legend-engine-test-runner-service/src/main/java/org/finos/legend/engine/testable/service/connection/TestConnectionBuilder.java

```diff
@@ -51,6 +51,10 @@ public class TestConnectionBuilder implements ConnectionVisitor<Pair<Connection,
 
     public TestConnectionBuilder(EmbeddedData embeddedData, PureModelContextData pureModelContextData)
     {
+        if (embeddedData == null)
```

Comment:
I don't think we can add this here, for example modelChainConnection code below don't require test data